docker/Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM alpine:edge
+
+EXPOSE 8080
+
+LABEL org.label-schema.name="laminar" \
+      org.label-schema.description="Fast and lightweight Continuous Integration" \
+      org.label-schema.usage="/usr/doc/UserManual.md" \
+      org.label-schema.url="https://laminar.ohwg.net" \
+      org.label-schema.vcs-url="https://github.com/ohwgiles/laminar" \
+      org.label-schema.schema-version="1.0" \
+      org.label-schema.docker.cmd="docker run -d -p 8080:8080 laminar"
+
+RUN apk add --no-cache -X http://dl-3.alpinelinux.org/alpine/edge/testing/ \
+        sqlite-dev \
+        zlib \
+        capnproto \
+        tini
+
+ADD UserManual.md /usr/doc/
+
+ADD . /build/laminar
+
+RUN apk add --no-cache --virtual .build -X http://dl-3.alpinelinux.org/alpine/edge/testing/ \
+        build-base \
+        cmake \
+        capnproto-dev \
+        boost-dev \
+        zlib-dev \
+        rapidjson-dev && \
+    cd /build/laminar && \
+    cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/ && \
+    make -j4 && \
+    make install && \
+    apk del .build && \
+    rm -rf /build
```

Comment:
Not too familiar with alpine. This looks cool, but I would have used a [multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/) for this. Thoughts?

Reply:
I chose this method over multi-stage build because of the external dependencies that are dynamically linked and must be present at build time and run time. This means the dependencies would have to be installed multiple times. Once in the build container and again in the final container. SQLite in particular, since by default laminar will not use the regular sqlite binaries, and requires the sqlite-dev package to be installed to run.

Not a huge problem, but the resulting image if really small already, and if the end user would like to optimize further they can use this image as a stage in their own multi-stage build, and pull the binaries into their own image.

Reply:
OK sure.

> SQLite in particular, since by default laminar will not use the regular sqlite binaries, and requires the sqlite-dev package to be installed to run

This sounds strange. Laminar should use the "regular" sqlite libraries at runtime, and only need sqlite-dev for building. At least this is how it works for Debian. Sounds like a bug, should be addressed separately.

Reply:
I figured it must be a configuration issue with the build on that platform. I can get the output from the container and open a new issue.

Reply:
@ohwgiles After a closer look at the output I found that Alpine's `sqlite` package does not contain the `.so` that laminar is trying to dynamically link. That is in another package `sqlite-libs` which is a dependency of `sqlite-dev`, but not `sqlite`. So I will be able to revisit this in the near future.

```
...
(75/76) Purging sqlite-libs (3.29.0-r0)
...
Error loading shared library libsqlite3.so.0: No such file or directory (needed by /usr/bin/laminard)
```

Reply:
OK cool. So I think just adding `sqlite-libs` where you have `zlib`, and moving `sqlite-dev` where you have `zlib-dev` should do the trick. Odd that in alpine it's split this way, it seems inconsistent with `zlib` and `capnproto`.